modin/core/storage_formats/base/query_compiler.py

```diff
@@ -4766,4 +4767,35 @@ def compare(self, other, align_axis, keep_shape, keep_equal, result_names):
             result_names=result_names,
         )
 
+    def repartition(self, axis):
+        """
+        Repartitioning QueryCompiler objects to get ideal partitions inside.
+
+        Allows to improve performance where the query compiler cannot yet.
+
+        Parameters
+        ----------
+        axis : int, optional
+
+        Returns
+        -------
+        BaseQueryCompiler
+            The repartitioned BaseQueryCompiler.
+        """
+        if StorageFormat.get() == "Hdk":
+            # Hdk uses only one partition, it makes
+            # no sense for it to repartition the dataframe.
+            return self
+
+        list_axis = [0, 1] if axis is None else [axis]
+
+        new_query_compiler = self
+        for _ax in list_axis:
+            new_query_compiler = new_query_compiler.__constructor__(
```

Comment:
Why does this line show that it is not covered by tests?

Reply:
Nice catch, this hasn't been tested in CI, fixed.